### PR TITLE
Created an amp-recaptcha-input example

### PR DESCRIPTION
**Pull request by torch2424**
This creates an example for the upcoming `amp-recaptcha-input`. However, this requires a backend, and is currently using [my glitch AMP recaptcha endpoint](https://glitch.com/edit/#!/torch2424-amp-glitch-express). Waiting for some direction on how to implement this in this project's express server

# Notes

cc @sebastianbenz

* Want me to open a PR in ABE as well? If anything, I can leave the ABE on glitch, and then write the glitch endpoint here?

* The experiment blurb was written by me. Please let me know if there is a new way of handling this. (The old way from ABE didn't work).

# Example

ampdevrecaptchaexample

### CODE REVIEW

File: examples/source/components/amp-recaptcha-input.html [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  <script async custom-element="amp-recaptcha-input" src="https://cdn.ampproject.org/v0/amp-recaptcha-input-0.1.js"></script>
+  <script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>
+  <script async custom-element="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+<style amp-custom>
+  form {
+    width: 100%;
+    height: 300px;
```
Comment (sebastianbenz) [inline examples/source/components/amp-recaptcha-input.html]: Can you avoid giving the form a fixed height? Or is this needed?

File: examples/source/components/amp-recaptcha-input.html [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      <label>
+        <span>Search for</span>
+        <input type="search" name="term" required>
+      </label>
+      <input name="submit-button" type="submit" value="Search">
+      <amp-recaptcha-input layout="nodisplay" name="recaptcha_token" data-sitekey="6LebBGoUAAAAAHbj1oeZMBU_rze_CutlbyzpH8VE" data-action="recaptcha_example">
+      </amp-recaptcha-input>
+    </fieldset>
+
+    <div class="loading-spinner">
```
Comment (sebastianbenz) [inline examples/source/components/amp-recaptcha-input.html]: A custom spinner is cool, but adds too much boilerplate to the sample. A simple message saying `Loading ...` is sufficient.

File: examples/source/components/amp-recaptcha-input.html [hunk 3]
```diff
@@ -0,0 +1,126 @@
+<!--
+  ## Experimental
```
[the remark this answers is not in the document]
Comment (torch2424) [reply examples/source/components/amp-recaptcha-input.html]: Ahhh I think the syntax for this changed from when I last did this? 🤔 Which is why I wrote my own cause it didn't work 😂  Either way, thank you!

[the remark this answers is not in the document]
Comment (torch2424) [reply examples/source/components/amp-recaptcha-input.html]: Oh actually! The issue is that you need to do:

`<!--- --->` and not `<!-- -->` (number of dashes).

According to MDN, this should be two dashes: https://developer.mozilla.org/en-US/docs/Web/API/Comment .

But that's just a nit by me, I think it should be mentioned somewhere (if not already)

**Comment by torch2424** [comment]
@sebastianbenz Made requested changes! Looks like CSS Vars aren't on the page though? 🤔 

Thanks for the review by the way! 😄

File: examples/source/components/amp-recaptcha-input.html [hunk 4]
```diff
@@ -14,7 +13,7 @@
 <head>
   <meta charset="utf-8">
   <title>amp-recaptcha-input</title>
-  <link rel="canonical" href="<%host%>/components/amp-recaptcha-input/">
+  <link rel="canonical" href="s/host/hosts.platform/components/amp-recaptcha-input/">
```
Comment (sebastianbenz) [inline examples/source/components/amp-recaptcha-input.html]: sorry for not being clear enough, this needs to be:

```
"<%host.platform%>/components/amp-recaptcha-input/"
```

Sidenote: I was using the old fashioned sed way to describe a change, the syntax means substitute OLD with NEW `s/OLD/NEW/`.

File: examples/source/components/amp-recaptcha-input.html [hunk 5]
```diff
@@ -24,62 +23,41 @@
   <script async custom-element="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
   <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
 <style amp-custom>
-  form {
+  #amp-recaptcha-input-form {
     width: 100%;
-    height: 300px;
-    overflow: auto;
+    padding: var(--space-1);
```
Comment (sebastianbenz) [inline examples/source/components/amp-recaptcha-input.html]: You need to manually add the variables to the file...

Comment (torch2424) [reply examples/source/components/amp-recaptcha-input.html]: Oh haha! My apologies on that. Thought it was a project wide variable 😂

**Comment by torch2424** [comment]
@sebastianbenz Made requested changes, this is good to go again!

My apologies on the huge delay, went on a roadshow, and handling a ton of High priority things lately.

**Fixed styles**

Screen Shot 2019-03-21 at 5 37 17 PM

**Comment by torch2424** [comment]
Also, we can take this offline, or open an issue. But if I may ask, how come we are including `:root` styles in all of the examples? I feel like copy pasting these could lead to a de-sync down the line. When we could have it templated / included in from a single place?

Thanks!

**Comment by sebastianbenz** [comment]
@torch2424 I get you're point, however this way sample files work standalone. The `:root` styles get automatically injected when you create a new sample using gulp.

**Comment by torch2424** [comment]
@sebastianbenz Removed the unused CSS vars as discussed offline. This is good to go! 😄

**Pull request merged**